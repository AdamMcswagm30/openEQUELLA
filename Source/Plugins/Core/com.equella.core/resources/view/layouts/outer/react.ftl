<#include "/com.tle.web.freemarker@/macro/sections/render.ftl"/><#t/>
<#assign template = m.template><#t/>
<!DOCTYPE html>
<html lang="${m.lang?html}" <#if m.rightToLeft>dir="rtl"</#if> ${m.htmlAttrs}>
    <head>
        <@render template["header"]/>
    	<script type="text/javascript">
    	var renderData = ${m.renderJs};
    	</script>
    </head>
    <body>
        <div id="mainDiv"></div>
        <script src="${m.scriptUrl}"></script>
        <@render template["postmarkup"]/>
    </body>
</html>
