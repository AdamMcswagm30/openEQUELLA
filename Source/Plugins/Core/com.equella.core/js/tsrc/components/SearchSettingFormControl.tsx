import {
  FormControl,
  FormControlLabel,
  FormHelperText,
  FormLabel
} from "@material-ui/core";
import * as React from "react";

export default function SearchSettingFormControl(props: {
  title?: string;
  label?: string;
  helperText?: string;
  disabled: boolean;
  control: React.ReactElement;
  onChange: (event: React.ChangeEvent<{}>, checked: boolean) => void;
}) {
  const { title, label, disabled, control, onChange, helperText } = props;
  return (
    <FormControl>
      {title && <FormLabel>{title}</FormLabel>}
      {helperText && <FormHelperText>{helperText}</FormHelperText>}
      <FormControlLabel
        disabled={disabled}
        label={label}
        control={control}
        onChange={onChange}
      />
    </FormControl>
  );
}
