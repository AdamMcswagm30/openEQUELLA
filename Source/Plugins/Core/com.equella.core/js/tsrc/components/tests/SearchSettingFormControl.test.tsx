import SearchSettingFormControl from "../SearchSettingFormControl";
import { Checkbox, TextField } from "@material-ui/core";
import * as React from "react";
import * as Adapter from "enzyme-adapter-react-16";
import { configure, shallow } from "enzyme";

configure({ adapter: new Adapter() });

test("snapshot test", () => {
  const onChange = jest.fn();
  const textbox = shallow(
    <SearchSettingFormControl
      control={
        <TextField id="testTextField" value={"test"} onChange={onChange} />
      }
      onChange={onChange}
    />
  );
  expect(textbox).toMatchSnapshot();
  const checkbox = shallow(
    <SearchSettingFormControl
      control={<Checkbox id="testCheckbox" value={true} onChange={onChange} />}
      title={"this one has a title"}
      label={"and a label"}
      helperText={"helper text"}
      onChange={onChange}
    />
  );
  expect(checkbox).toMatchSnapshot();
});
