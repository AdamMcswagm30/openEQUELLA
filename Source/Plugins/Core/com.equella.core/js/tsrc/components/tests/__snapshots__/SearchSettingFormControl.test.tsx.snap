// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<WithStyles(FormControl)>
  <WithStyles(WithFormControlContext(FormControlLabel))
    control={
      <TextField
        id="testTextField"
        onChange={[MockFunction]}
        required={false}
        select={false}
        value="test"
        variant="standard"
      />
    }
    onChange={[MockFunction]}
  />
</WithStyles(FormControl)>
`;

exports[`snapshot test 2`] = `
<WithStyles(FormControl)>
  <WithStyles(WithFormControlContext(FormLabel))>
    this one has a title
  </WithStyles(WithFormControlContext(FormLabel))>
  <WithStyles(WithFormControlContext(FormHelperText))>
    helper text
  </WithStyles(WithFormControlContext(FormHelperText))>
  <WithStyles(WithFormControlContext(FormControlLabel))
    control={
      <WithStyles(Checkbox)
        id="testCheckbox"
        onChange={[MockFunction]}
        value={true}
      />
    }
    label="and a label"
    onChange={[MockFunction]}
  />
</WithStyles(FormControl)>
`;
